files/bootstrap.sh

```diff
@@ -481,6 +481,13 @@ if vercmp "$KUBELET_VERSION" lt "1.26.0"; then
 else
   KUBELET_CLOUD_PROVIDER="external"
   echo "$(jq ".providerID=\"$(provider-id)\"" $KUBELET_CONFIG)" > $KUBELET_CONFIG
+  # When the external cloud provider is used, kubelet will use /etc/hostname as the name of the Node object.
+  # If the VPC has a custom `domain-name` in its DHCP options set, and the VPC has `enableDnsHostnames` set to `true`,
+  # then /etc/hostname is not the same as EC2's PrivateDnsName.
+  # The name of the Node object must be equal to EC2's PrivateDnsName for the aws-iam-authenticator to allow this kubelet to manage it.
+  INSTANCE_ID=$(imds /latest/meta-data/instance-id)
+  PRIVATE_DNS_NAME=$(aws ec2 describe-instances --instance-ids $INSTANCE_ID --query 'Reservations[].Instances[].PrivateDnsName' --output text)
```

Comment:
why not use IMDS instead?

```
imds latest/meta-data/local-hostname
i-00ccc01fc3fcd3e27.us-east-2.compute.internal
```
?

Reply:
`local-hostname` doesn't always match EC2's `PrivateDnsName`.

If:
1. A custom domain is configured on the VPC. (e.g. `foo`)
2. The VPC has `enableDnsHostnames` enabled

Then:
- `local-hostname` will be `$NAME.foo` (`NAME` is either resource-based or IP-based).
- `PrivateDnsName` will be `$NAME.$REGION.compute.internal`.